pkg/resource/resource.go

```diff
@@ -243,45 +243,33 @@ func IsConditionTrue(c v1alpha1.Condition) bool {
 
 // An Applicator applies changes to an object.
 type Applicator interface {
-	Apply(context.Context, client.Client, runtime.Object, ...ApplyOption) error
+	Apply(context.Context, client.Client, runtime.Object, ...ApplyOptFn) error
 }
 
 // An ApplyFn is a function that satisfies the Applicator interface.
-type ApplyFn func(context.Context, client.Client, runtime.Object, ...ApplyOption) error
+type ApplyFn func(context.Context, client.Client, runtime.Object, ...ApplyOptFn) error
 
 // Apply changes to the supplied object.
-func (fn ApplyFn) Apply(ctx context.Context, c client.Client, o runtime.Object, ao ...ApplyOption) error {
+func (fn ApplyFn) Apply(ctx context.Context, c client.Client, o runtime.Object, ao ...ApplyOptFn) error {
 	return fn(ctx, c, o, ao...)
 }
 
-// ApplyOptions configure how changes are applied to an object.
-type ApplyOptions struct {
-	// ControllersMustMatch requires any existing object to have a controller
-	// reference, and for that controller reference to match the controller
-	// reference of the supplied object.
-	ControllersMustMatch bool
-}
-
-// An ApplyOption configures how changes are applied to an object.
-type ApplyOption func(a *ApplyOptions)
+// An ApplyOptFn is a function that checks for a condition before patch.
+type ApplyOptFn func(ctx context.Context, current, desired runtime.Object) error
```

Comment:
I'd prefer to keep the name `ApplyOption` here, despite our common pattern of pluggable functions ending in `Fn`.

Reply:
Also, how likely do you think we are to need `ctx` here?

Reply:
I would think not very likely, but why not pass it in case?

Reply:
Haha okay I'm sold. Can you change it to `_`in `ControllersMustMatch` to reflect that it's unused?

Reply:
Done :)